components/b2b/AccountHierarchy/AccountHierarchyTree/AccountHierarchyTree.tsx

```diff
@@ -3,23 +3,33 @@ import * as React from 'react'
 import ChevronRightIcon from '@mui/icons-material/ChevronRight'
 import DragIndicator from '@mui/icons-material/DragIndicator'
 import ExpandMoreIcon from '@mui/icons-material/ExpandMore'
-import { Box, Button, IconButton, ListItemIcon } from '@mui/material'
+import {
+  Box,
+  Button,
+  IconButton,
+  ListItemIcon,
+  Typography,
+  useMediaQuery,
+  useTheme,
+} from '@mui/material'
 import { useTranslation } from 'next-i18next'
 import Nestable from 'react-nestable'
 
+import { AccountHierarchyStyles } from './AccountHierarchyTree.styles'
 import { AccountHierarchyTreeLabel } from '@/components/b2b'
-interface Hierarchy {
-  id: string | number
-  children: Hierarchy[]
-}
+import { AddChildAccountProps, EditChildAccountProps, HierarchyNode } from '@/lib/types'
 
-interface TreeItemListProps {
-  accounts: any[]
-  hierarchy: Hierarchy
-}
+import { B2BAccount } from '@/lib/gql/types'
 
-interface AccountHierarchyTreeProps extends TreeItemListProps {
+interface AccountHierarchyTreeProps {
   role: string
+  accounts: any[]
+  hierarchy: HierarchyNode[] | undefined
+  handleViewAccount: (item: B2BAccount) => void
+  handleAddAccount: ({ isAddingAccountToChild, accounts }: AddChildAccountProps) => void
+  handleEditAccount: ({ accounts }: EditChildAccountProps) => void
+  handleSwapAccount: (b2BAccount: B2BAccount) => void
+  handleDeleteAccount: (b2BAccount: B2BAccount) => void
```

Comment:
As there is no Delete functionality rename to handleDisableAccount: (b2BAccountId: string) => void

Reply:
Done